From 90965cded0bf68238ec0b7308bfddf2945177fef Mon Sep 17 00:00:00 2001
From: Stefan Berger <stefanb@linux.ibm.com>
Date: Mon, 24 Jan 2022 09:56:45 -0500
Subject: [PATCH 15/51] ima: Check IMA_MEASURED flag using flags from
 iint_flags()

Use iint_flags() to get the flag when checking whether IMA_MEASURED is
set.

Signed-off-by: Stefan Berger <stefanb@linux.ibm.com>
---
 security/integrity/ima/ima_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/integrity/ima/ima_main.c b/security/integrity/ima/ima_main.c
index c6fdd1277795..0fb7ddb743dd 100644
--- a/security/integrity/ima/ima_main.c
+++ b/security/integrity/ima/ima_main.c
@@ -375,7 +375,7 @@ static int __process_measurement(struct ima_namespace *ns,
 			rc = ima_read_modsig(func, buf, size, &modsig);
 
 			if (!rc && ima_template_has_modsig(template_desc) &&
-			    iint->flags & IMA_MEASURED)
+			    iint_flags(iint, ns_status) & IMA_MEASURED)
 				action |= IMA_MEASURE;
 		}
 	}
-- 
2.31.1

